stone/lang/parser.py

```diff
@@ -982,6 +983,46 @@ def p_ex_list_items_extend(self, p):
         p[0].append(p[3])
 
     # --------------------------------------------------------------
+    # Maps
+    #
+
+    def p_ex_map(self, p):
+        """ex_map : LBRACE ex_map_pairs RBRACE
+                  | LBRACE empty RBRACE"""
+        p[0] = {} if p[2] is None else p[2]
+
+    def p_ex_map_elem_primitive(self, p):
+        """ex_map_elem : primitive"""
+        p[0] = None if p[1] == StoneNull else p[1]
+
+    def p_ex_map_elem_composit(self, p):
+        """ex_map_elem : ex_map
+                       | ex_list"""
+        p[0] = p[1]
```

Comment:
My understanding was that by deferring the conversion (at least in the `ex_list` case), then I'm not duplicating conversion code in the parser.